src/AggregateFunctions/DDSketch.h

```diff
@@ -0,0 +1,239 @@
+            if (mapping->getGamma() > other.mapping->getGamma())
+            {
+                BaseQuantileDDSketch new_sketch = other.changeMapping(mapping->getGamma());
+                this->merge(new_sketch);
+                return;
+            }
+            else
+            {
+                BaseQuantileDDSketch new_sketch = changeMapping(other.mapping->getGamma());
+                copy(new_sketch);
+            }
+        }
+
+        // If the other sketch is empty, do nothing
+        if (other.count == 0)
+        {
+            return;
+        }
+
+        // If this sketch is empty, copy the other sketch
+        if (count == 0)
+        {
+            copy(other);
+            return;
+        }
+
+        count += other.count;
+        zero_count += other.zero_count;
+
+        store->merge(other.store.get());
+        negative_store->merge(other.negative_store.get());
+    }
+
+    void serialize(WriteBuffer& buf) const
+    {
+        // Write the mapping
+        writeBinary(enc.FlagIndexMappingBaseLogarithmic.byte, buf);
+        mapping->serialize(buf);
+
+        // Write the positive and negative stores
+        writeBinary(enc.FlagTypePositiveStore, buf);
+        store->serialize(buf);
+
+        writeBinary(enc.FlagTypeNegativeStore, buf);
+        negative_store->serialize(buf);
+
+        // Write the zero count
+        writeBinary(enc.FlagZeroCountVarFloat.byte, buf);
+        writeBinary(zero_count, buf);
+    }
+
+    void deserialize(ReadBuffer& buf)
+    {
+        // Read the mapping
+        UInt8 flag = 0;
+        readBinary(flag, buf);
+        if (flag != enc.FlagIndexMappingBaseLogarithmic.byte)
+        {
+            throw Exception(ErrorCodes::INCORRECT_DATA, "Invalid flag for mapping");
+        }
+        mapping->deserialize(buf);
+
+        // Read the positive and negative stores
+        readBinary(flag, buf);
+        if (flag != enc.FlagTypePositiveStore)
+        {
+            throw Exception(ErrorCodes::INCORRECT_DATA, "Invalid flag for positive store");
+        }
+        store->deserialize(buf);
+
+        readBinary(flag, buf);
+        if (flag != enc.FlagTypeNegativeStore)
+        {
+            throw Exception(ErrorCodes::INCORRECT_DATA, "Invalid flag for negative store");
+        }
+        negative_store->deserialize(buf);
+
+        // Read the zero count
+        readBinary(flag, buf);
+        if (flag != enc.FlagZeroCountVarFloat.byte)
+        {
+            throw Exception(ErrorCodes::INCORRECT_DATA, "Invalid flag for zero count");
+        }
+        readBinary(zero_count, buf);
+        count = static_cast<Float64>(negative_store->count + zero_count + store->count);
+    }
+
+private:
+    std::unique_ptr<KeyMapping> mapping;
+    std::unique_ptr<DenseStore> store;
+    std::unique_ptr<DenseStore> negative_store;
+    Float64 zero_count;
+    Float64 count;
+    DDSketchEncoding enc;
+    Poco::Logger * log = &Poco::Logger::get("DDSketch");
```

Comment:
Unused? Also, doesn't belong here, better to create it when something needs to be logger rather than for every row of aggregation results - the latter can be much more frequent (and the former is amortized by the cost of logging a message).